### DISCUSSION TITLE
Unacceptable Support for Linux

### DISCUSSION THREAD

**Discussion by darkoverlordofdata** [unoplatform/uno#11800] [General]
I suspect this is induced by your templates, see my prior issue #11753 . 
When I try your demo’s on github (unoplatform/uno.Samples, unoplatform.uno.todo) on Linux, they do not work well. 
* I get a prolonged black screen at startup. 
* The general performance is poor. 
* Components (text boxes, buttons, etc) flicker and jiggle. 
* In some cases, the keyboard stops working when I exit the program, and the only way to recover from this is a hard/power reset. 
In general, these are unacceptable, I would not release an application targeting Linux based on Uno in its current state.

Then I found this tutorial - (https://steven-giesel.com/blogPost/b2234ada-0978-4c7b-841e-ca6a255247b0) which has none of these issues - it’s actually impressive to see. When I compare the *.Skia.Grk.csproj  on these projects, I see that the project which works references a different set of components - it uses Fluent design components. The problem with that is these are Windows style, not what anyone expects to see on Linux.

When I try simply “dotnet new unoapp -o MyApp”, I do not get the Fluent components, and MyApp has the same problems. It seems that to get a working Linux app, I need to clone https://github.com/linkdotnet/BlogExamples/tree/main/TodoApp, and strip out the original code, adding my own code back in. So I would say that the dev experience is also unacceptable.

The amount of work it would take to get professional looking results on Linux really makes Uno a non-starter for me.

**Comment by FrozDark** [comment]
Almost every platform is nearly unacceptable cause of basic bugs with the collections control and binding issues that nobody is resolving for a long time
You have to use workarounds everywhere

P.S.: Uno is commercial. Only paid work would be done

**Comment by jeromelaban** [comment]
@darkoverlordofdata thanks for your feedback.

It seems you're hitting a series of small issues on various scenarios, which generally need individual bug reports. 

Here's some feedback on some:

> * I get a prolonged black screen at startup.

This is likely related to the .NET JIT starting up. Depending on the device you're using it's probably best to use either [Ready To Run](https://learn.microsoft.com/en-us/dotnet/core/deploying/ready-to-run) or [.NET Native](https://platform.uno/docs/articles/features/using-skia-gtk.html?q=gtk#net-native-aot-support) to get better performance.

> * The general performance is poor.

What device do you use? You may be running with [software rendering](https://platform.uno/docs/articles/features/using-skia-gtk.html?q=gtk#troubleshooting-opengl-integration) that is generally slower.

> * Components (text boxes, buttons, etc) flicker and jiggle.

Can you open an issue with a video and repro, if possible?

> * In some cases, the keyboard stops working when I exit the program, and the only way to recover from this is a hard/power reset.

You're likely using framebuffer apps. If you tried some our samples, it may be that [this block is missing](https://github.com/unoplatform/uno.extensions/blob/8d491ea360dda7721afc9bf872dc5fe6db35bc68/src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.1.Skia.Linux.FrameBuffer/Program.cs#L18-L32).

> Then I found this tutorial - (https://steven-giesel.com/blogPost/b2234ada-0978-4c7b-841e-ca6a255247b0) which has none of these issues - it’s actually impressive to see. When I compare the *.Skia.Grk.csproj on these projects, I see that the project which works references a different set of components - it uses Fluent design components. The problem with that is these are Windows style, not what anyone expects to see on Linux.

You've been trying samples built with the previous set of samples that were using what is now called the "blank" preset. The new default template uses the Material theme. If you wish to have a blank app, you'll need to use [`--preset=blank`](https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=net7%2Cwindows#uno-platform-application).

> In general, these are unacceptable, I would not release an application targeting Linux based on Uno in its current state.
> The amount of work it would take to get professional looking results on Linux really makes Uno a non-starter for me.

We're mainly working based on our clients needs first, this is [how we sustain Uno Platform](https://platform.uno/blog/sustaining-the-open-source-uno-platform/). If you wish to get your issues addressed faster, give us a hello at info@platform.uno!

**Comment by darkoverlordofdata** [comment]
Thank you - You are correct, I apologize for any flame, I was having a
migraine, I do not communicate well when this happens.

The bug I was referring to is this:
https://github.com/unoplatform/uno/issues/11066

I find this bug gives me a migraine, it gave me the migraine I was
struggling with the other day.

There is a workaround - use an older release - I found this project (
https://steven-giesel.com/blogPost/b2234ada-0978-4c7b-841e-ca6a255247b0) -
it is using a release from prior year. I can then upgrade when the bug is
corrected.


What I am struggling with is how to find this intentionally - for example
rolling back incrementally until the bug is gone. Is there a way of doing
that?


On Mon, Mar 27, 2023 at 5:12 AM Jérôme Laban ***@***.***>
wrote:

> @darkoverlordofdata <https://github.com/darkoverlordofdata> thanks for
> your feedback.
>
> It seems you're hitting a series of small issues on various scenarios,
> which generally need individual bug reports.
>
> Here's some feedback on some:
>
>
>    - I get a prolonged black screen at startup.
>
> This is likely related to the .NET JIT starting up. Depending on the
> device you're using it's probably best to use either Ready To Run
> <https://learn.microsoft.com/en-us/dotnet/core/deploying/ready-to-run> or .NET
> Native
> <https://platform.uno/docs/articles/features/using-skia-gtk.html?q=gtk#net-native-aot-support>
> to get better performance.
>
>
>    - The general performance is poor.
>
> What device do you use? You may be running with software rendering
> <https://platform.uno/docs/articles/features/using-skia-gtk.html?q=gtk#troubleshooting-opengl-integration>
> that is generally slower.
>
>
>    - Components (text boxes, buttons, etc) flicker and jiggle.
>
> Can you open an issue with a video and repro, if possible?
>
>
>    - In some cases, the keyboard stops working when I exit the program,
>    and the only way to recover from this is a hard/power reset.
>
> You're likely using framebuffer apps. If you tried some our samples, it
> may be that this block is missing
> <https://github.com/unoplatform/uno.extensions/blob/8d491ea360dda7721afc9bf872dc5fe6db35bc68/src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.1.Skia.Linux.FrameBuffer/Program.cs#L18-L32>
> .
>
> Then I found this tutorial - (
> https://steven-giesel.com/blogPost/b2234ada-0978-4c7b-841e-ca6a255247b0)
> which has none of these issues - it’s actually impressive to see. When I
> compare the *.Skia.Grk.csproj on these projects, I see that the project
> which works references a different set of components - it uses Fluent
> design components. The problem with that is these are Windows style, not
> what anyone expects to see on Linux.
>
> You've been trying samples built with the previous set of samples that
> were using what is now called the "blank" preset. The new default template
> uses the Material theme. If you wish to have a blank app, you'll need to
> use --preset=blank
> <https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=net7%2Cwindows#uno-platform-application>
> .
>
> In general, these are unacceptable, I would not release an application
> targeting Linux based on Uno in its current state.
> The amount of work it would take to get professional looking results on
> Linux really makes Uno a non-starter for me.
>
> We're mainly working based on our clients needs first, this is how we
> sustain Uno Platform
> <https://platform.uno/blog/sustaining-the-open-source-uno-platform/>. If
> you wish to get your issues addressed faster, give us a hello at
> ***@***.***!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unoplatform/uno/discussions/11800#discussioncomment-5440211>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJUE4TV3H7QEMHBW3UHJJLW6F74XANCNFSM6AAAAAAWHSLU5U>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Bruce Davidson

**Reply by jeromelaban** [reply]
Thanks for the update on the issue. Indeed we're working to get it fixed, though it may be specifically related to the use of material. If you use fluent styles only (by removing any material related entries in `appresources.xaml` or `app.xaml` if any), this may not be a problem. Have you tried this?

**Comment by darkoverlordofdata** [comment]
no, I didn't realize I could do that. It works, and that's a better
solution, easy to revert once the bug is fixed. Thank you!

On Tue, Mar 28, 2023 at 7:56 AM Jérôme Laban ***@***.***>
wrote:

> Thanks for the update on the issue. Indeed we're working to get it fixed,
> though it may be specifically related to the use of material. If you use
> fluent styles only (by removing any material related entries in
> appresources.xaml or app.xaml if any), this may not be a problem. Have
> you tried this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unoplatform/uno/discussions/11800#discussioncomment-5455307>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJUE4VXJLEOHOWSRZYZVN3W6L32JANCNFSM6AAAAAAWHSLU5U>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Bruce Davidson
